test/unit/Reflection/ReflectionObjectTest.php

```diff
@@ -106,6 +106,19 @@ public function testGetRuntimePropertiesWithFilter() : void
         self::assertEmpty($classInfo->getProperties(\ReflectionProperty::IS_PRIVATE));
     }
 
+    public function testGetRuntimeImmediatePropertiesWithFilter() : void
+    {
+        $foo = new \StdClass();
+        $foo->bar = 'huzzah';
+
+        $classInfo = ReflectionObject::createFromInstance($foo);
+
+        self::assertEmpty($classInfo->getImmediateProperties(\ReflectionProperty::IS_STATIC));
+        self::assertCount(1, $classInfo->getImmediateProperties(\ReflectionProperty::IS_PUBLIC));
```

Comment:
Uhm, I am not sure if this should be in the same reflection API, since runtime access will fail when done via normal reflection.

Should probably be a separate API endpoint.